core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -530,33 +541,16 @@ private void tryStart(Instant startedAt) {
     @VisibleForTesting
     Checksum hashCopiedFiles() {
         Checksum checksum = new Adler32();
-        copyToFileContainerPathMap.entrySet().stream().sorted(Entry.comparingByValue()).forEach(entry -> {
-            byte[] pathBytes = entry.getValue().getBytes();
-            // Add path to the hash
-            checksum.update(pathBytes, 0, pathBytes.length);
-
-            File file = new File(entry.getKey().getResolvedPath());
-            checksumFile(file, checksum);
-        });
+        Stream.of(copyToFileContainerPathMap, copyToTransferableContainerPathMap)
+            .flatMap(it -> it.entrySet().stream())
+            .sorted(Entry.comparingByValue()).forEach(entry -> {
+                byte[] bytes = entry.getValue().getBytes();
+                checksum.update(bytes, 0, bytes.length);
```

Comment:
nit: let's make the changelog smaller and restore the comment:
```suggestion
                byte[] pathBytes = entry.getValue().getBytes();
                // Add path to the hash
                checksum.update(pathBytes, 0, pathBytes.length);

```